SimTKcommon/include/SimTKcommon/internal/IteratorRange.h

```diff
@@ -0,0 +1,62 @@
+#ifndef SimTK_SimTKCOMMON_ITERATOR_RANGE_H_
+#define SimTK_SimTKCOMMON_ITERATOR_RANGE_H_
+
+/* -------------------------------------------------------------------------- *
+ *                       Simbody(tm): SimTKcommon                             *
+ * -------------------------------------------------------------------------- *
+ * This is part of the SimTK biosimulation toolkit originating from           *
+ * Simbios, the NIH National Center for Physics-Based Simulation of           *
+ * Biological Structures at Stanford, funded under the NIH Roadmap for        *
+ * Medical Research, grant U54 GM072970. See https://simtk.org/home/simbody.  *
+ *                                                                            *
+ * Portions copyright (c) 2008-16 Stanford University and the Authors.        *
+ * Authors: Chris Dembia                                                      *
+ * Contributors:                                                              *
+ *                                                                            *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may    *
+ * not use this file except in compliance with the License. You may obtain a  *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.         *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing, software        *
+ * distributed under the License is distributed on an "AS IS" BASIS,          *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ * -------------------------------------------------------------------------- */
+
+namespace SimTK {
+
+/** Helper class to use range-based for loops with a pair of iterators. This
+ * class should only be used when you're sure the iterators are valid. Don't
+ * use this class directly; instead, use makeIteratorRange().
+ *
+ * @code
+ * for (auto& x : makeIteratorRange(v.begin(), v.end())) {
+ *     ...
+ * }
+ * @endcode
+ * */
```

Comment:
Style comment, feel free to ignore: in Simbody I don't use the initial ` * ` for multiline comments. Because all modern code viewers colorize comments, that is no longer necessary to make them visible. It is an anachronism from when editors were black and white. Getting rid of those gives back 3 characters from our 80 character budget, and reduces clutter IMO. (Exception: the copyright block at the top of every file which is intentionally drawn in a box.)